helm-charts/splunk-otel-collector/templates/daemonset.yaml

```diff
@@ -124,6 +124,45 @@ spec:
               mountPath: /var/log
             - name: varlibdockercontainers
               mountPath: /var/lib/docker/containers
+        {{- if .Values.otelAgent.runInitForNonRoot }}
+        - name: chown
+          image: registry.access.redhat.com/ubi8/ubi
+          command: ['sh', '-c', '
+          mkdir -p {{ .Values.logsCollection.checkpointPath }};
+          chown -Rv {{ .Values.otelAgent.securityContext.runAsUser | default 20000 }}:{{ .Values.otelAgent.securityContext.runAsGroup | default 20000 }} {{ .Values.logsCollection.checkpointPath }};
+          chmod -v g+rwxs {{ .Values.logsCollection.checkpointPath }};
+          {{ if .Values.logsCollection.containers.enabled -}}
+          if [ -d "/var/lib/docker/containers" ];
+          then
+              chgrp -Rv {{ .Values.otelAgent.securityContext.runAsGroup | default 20000 }} /var/lib/docker/containers;
+              chmod -R g+rwxs /var/lib/docker/containers;
```

Comment:
Why do we need to grant write permissions here? We should never write to these directories, only to checkpoints directory

Reply:
And how will it work for new log files? This init container recursively patches existing log files only once, but new log files (from new pods or rotated files) won't be patched and collector won't be able to read them. Or I'm missing something?

May be some linux capabilities can give us a way to read these files instead of patching them...

Reply:
right. we don't need write permission. will update it.

Reply:
`setfacl` command takes care of new files added to this directory.